### PR TITLE
feat(aws): Adding support for update asg launch template

### PR DESCRIPTION
**Pull request by jeyrschabu**
- modify server group launch template saga support
- this operation will be used by rollingpush and the editing of asg's security groups

### CODE REVIEW

**Comment by jeyrschabu** [comment]
Reintroduce https://github.com/spinnaker/clouddriver/commit/b5efd0d90ff25eb153caf9e849ae7203f94ba1d3 with a fix to the issue that led to the revert.

When sql.enabled=false and aws is the provider.

File: clouddriver-event/src/main/kotlin/com/netflix/spinnaker/clouddriver/config/EventConfiguration.kt [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+ * limitations under the License.
+ */
+
+package com.netflix.spinnaker.clouddriver.config
+
+import org.springframework.context.annotation.ComponentScan
+import org.springframework.context.annotation.Configuration
+
+@Configuration
+@ComponentScan(basePackages = ["com.netflix.spinnaker.clouddriver.event"])
```
Comment (jeyrschabu) [inline clouddriver-event/src/main/kotlin/com/netflix/spinnaker/clouddriver/config/EventConfiguration.kt:23]: The in-memory-event repository was previously not seen by Spring. This fixes that.

**Comment by jeyrschabu** [comment]
Can I get a thumbs up on this? Would like to test this out this am.

**Review by caseyhebebrand — APPROVED** [review]
The launch template logic looks good to me, but I think someone who is more familiar with sagas should also take a look.

File: clouddriver-event/src/main/kotlin/com/netflix/spinnaker/clouddriver/config/EventConfiguration.kt (hunk 1 above)
Comment (jeyrschabu) [reply clouddriver-event/src/main/kotlin/com/netflix/spinnaker/clouddriver/config/EventConfiguration.kt:23]: FYI @robzienert

**Comment by jeyrschabu** [comment]
Saga parts are the same as last PR with the only addition being the fix for spring to be able to find the in memory event repo.

Merging for now but feel free to comment on things and I will address them in separate PRs.

**Pull request merged**